autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/FastICA.py

```diff
@@ -0,0 +1,102 @@
+from math import ceil, floor
+from typing import Any, Dict, List, Optional
+
+from ConfigSpace.conditions import EqualsCondition
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformIntegerHyperparameter,
+)
+
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.decomposition import FastICA as SklearnFastICA
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import HyperparameterSearchSpace, add_hyperparameter, get_hyperparameter
+
+
+class FastICA(autoPyTorchFeaturePreprocessingComponent):
+    def __init__(self, n_components: Optional[int] = None,
+                 algorithm: str = 'euclidean', whiten: str = 'ward',
```

Comment:
[The documentation](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.FastICA.html) says `whiten` is bool?
And `algorithm` is `parallel` or `deflation`?

Reply:
thanks for pointing out, I totally messed this file